_studio/mfx_lib/decode/av1/src/mfx_av1_dec_decode.cpp

```diff
@@ -1149,10 +1162,16 @@ inline void CopyFilmGrainParam(mfxExtAV1FilmGrainParam &extBuf, UMC_AV1_DECODER:
         extBuf.Flags |= MFX_FILM_GRAIN_CLIP_TO_RESTRICTED_RANGE;
 }
 
-mfxStatus VideoDECODEAV1::FillOutputSurface(mfxFrameSurface1** surf_out, mfxFrameSurface1* surface_work, UMC_AV1_DECODER::AV1DecoderFrame* pFrame)
+mfxStatus VideoDECODEAV1::FillOutputSurface(mfxBitstream *bs, mfxFrameSurface1** surf_out, mfxFrameSurface1* surface_work, UMC_AV1_DECODER::AV1DecoderFrame* pFrame)
 {
     MFX_CHECK_NULL_PTR3(surface_work, pFrame, surf_out);
 
+    if (!pFrame->GetShowFrame())
+    {
+        surf_out = 0;
+        return (mfxStatus)MFX_ERR_MORE_DATA_SUBMIT_TASK;
+    }
+
```

Comment:
It should be handled in GetFrameToDisplay(), not in the FillOutputSurface function which just adds some meta info to mfxFrameSurface1

Reply:
Yes. Its has been checked in the following:
https://github.com/Intel-Media-SDK/MediaSDK/blob/master/_studio/shared/umc/codec/av1_dec/src/umc_av1_decoder.cpp#L546
I have removed the section in the latest commit.
Thanks.